tests/test_material.py

```diff
@@ -0,0 +1,57 @@
+import pytest
+
+from hexrd.material import Material
+
+# Tolerance for comparing floats
+FLOAT_TOL = 1.e-8
+
+# Use consistent units to simplify testing
+DEFAULT_LENGTH_UNIT = 'angstrom'
+DEFAULT_ANGLE_UNIT = 'degrees'
+
+
+@pytest.fixture
+def default_material():
+    return Material()
+
+
+def normalize_unit(v):
+    if hasattr(v, 'unit'):
+        # Assume it's a val unit
+        if v.unit in ('radians', 'degrees'):
+            # Assume it's an angle
+            return v.getVal(DEFAULT_ANGLE_UNIT)
+        else:
+            # Assume it's a length
+            return v.getVal(DEFAULT_LENGTH_UNIT)
+
+    return v
+
+
+def are_close(v1, v2, tol=FLOAT_TOL):
+    return abs(normalize_unit(v1) - normalize_unit(v2)) < tol
+
+
+def lparms_are_close(lparms, indices, tol=FLOAT_TOL):
+    first = lparms[indices[0]]
+    return all(are_close(first, lparms[i], tol) for i in indices[1:])
+
+
+def test_sgnum_setter(default_material):
```

Comment:
Just to explain what is going on here, pytest does a few things automatically based upon how you name things.

If you start a function name with `test_`, pytest assumes it is a test and tries to run it.

Arguments that you put in the `test_` function can be pytest fixtures. In this case, the argument is named `default_material`, which, via name matching, pytest automatically associates with [the fixture function defined above named `default_material`](https://github.com/HEXRD/hexrd/pull/220/files#diff-428844a4010d0b1e799aacd172b9ac423226cd6ce830c4c75858b434be6df8b5R13-R15). It automatically calls that function and passes the returned value as the argument to this test function.